### ISSUE TITLE
Why don't log var and mu appear in your code? Is the loss function correct?

### ISSUE THREAD

**Issue by CXX1113** [Skydes/Concrete-Dropout#2]
Hi @Skydes , 

1. Line 36 in mnist.py,  you didn't figure out log var and mu, which is different from the original code implementation.
2. I'm glad to see that your loss is the loss for the classification task. Because the original code only provides the loss of the regression task. But why doesn't your classification loss have KL loss about posttest?  (the regular loss in your loss is only a priori KL term)

**Comment by sarlinpe** [comment]
1. This codebase is not intended to be an exact replication of the original implementation, but rather a cleaner example on how to apply concrete dropout. As such, the network does not predict the aleatoric uncertainty (log var), which is anyway defined for regression but not classification.
2. The dropout rate regularizer is [applied](https://github.com/Skydes/Concrete-Dropout/blob/master/concrete_dropout.py#L119) in the ConcreteDropout layers, while the kernel regularizer is [passed](https://github.com/Skydes/Concrete-Dropout/blob/master/mnist.py#L21) to the convolution layers. Both are are gathered in the collection `tf.GraphKeys.REGULARIZATION_LOSSES`, and [applied to the loss](https://github.com/Skydes/Concrete-Dropout/blob/master/mnist.py#L52).

**Comment by CXX1113** [comment]
Thank you for your reply. :)
1. Why the classification task does not need log var, can you give me a hint? Because the original paper didn't describe the purpose of classification about concrete dropout in detail, which confused me very much.
2.  I thought that kernel regularizer and dropout rate regularizer are just priors of KL (it have been said that in https://github.com/yaringal/ConcreteDropout/blob/master/concrete-dropout-keras.ipynb), not posterior distributions included. (i.e., not complete KL term) . I'm not sure if my idea is right.

**Comment by sarlinpe** [comment]
1. For classification the Softmax entropy averaged over stochastic passes can already represent the spread of the multinational distribution unconditioned on the weight - the second term of Equation 2 in our [benchmark paper](https://arxiv.org/abs/1904.03215).

2. Afaik priors of KL are sufficient.
